File: app/src/main/java/al/bruno/financaime/util/EventDecorator.java
package al.bruno.financaime.util;

import com.prolificinteractive.materialcalendarview.CalendarDay;
import com.prolificinteractive.materialcalendarview.DayViewDecorator;
import com.prolificinteractive.materialcalendarview.DayViewFacade;
import com.prolificinteractive.materialcalendarview.spans.DotSpan;

import java.util.Date;
import java.util.HashSet;
import java.util.List;

public class EventDecorator implements DayViewDecorator {
    private final int color;
    private final HashSet<CalendarDay> dates;

    public EventDecorator(int color, /*Collection<CalendarDay> dates*/List<Date> dateList) {

        this.dates = new HashSet<>();
        for(Date date : dateList){
            dates.add(CalendarDay.from(date));
        }
        this.color = color;
    }

    @Override
    public boolean shouldDecorate(CalendarDay day) {
        return dates.contains(day);
    }

    @Override
    public void decorate(DayViewFacade view) {
        view.addSpan(new DotSpan(5, color));
    }
}

File: app/src/main/java/al/bruno/personal/expense/util/EventDecorator.kt
package al.bruno.personal.expense.util

import com.prolificinteractive.materialcalendarview.CalendarDay
import com.prolificinteractive.materialcalendarview.DayViewDecorator
import com.prolificinteractive.materialcalendarview.DayViewFacade
import com.prolificinteractive.materialcalendarview.spans.DotSpan

import org.threeten.bp.DateTimeUtils
import org.threeten.bp.ZoneId

import java.util.Date
import java.util.HashSet

class EventDecorator(private val color: Int, /*Collection<CalendarDay> dates*/dateList: List<Date>) : DayViewDecorator {
    private val dates: HashSet<CalendarDay>

    init {
        this.dates = HashSet()
        for (date in dateList) {
            dates.add(CalendarDay.from(DateTimeUtils.toInstant(date).atZone(ZoneId.systemDefault()).toLocalDate()))
        }
    }

    override fun shouldDecorate(day: CalendarDay): Boolean {
        return dates.contains(day)
    }

    override fun decorate(view: DayViewFacade) {
        view.addSpan(DotSpan(5f, color))
    }
}
